react-app/src/library/services/headset.ts

```diff
@@ -0,0 +1,238 @@
+      this.logger.info('Headset: No vendor headset connected [outgoingCall]');
+      return Promise.resolve();
+    }
+
+    return service.outgoingCall(callInfo);
+  }
+
+  answerCall(conversationId: string): Promise<any> {
+    const service = this.selectedImplementation;
+    if (!service || !service.isConnected) {
+      this.logger.info('Headset: No vendor headset connected [answerCall]');
+      return Promise.resolve();
+    }
+    return service.answerCall(conversationId);
+  }
+
+  setMute(value: boolean): Promise<any> {
+    const service = this.selectedImplementation;
+    if (!service || !service.isConnected) {
+      this.logger.info('Headset: No venddor headset connected [setMute]');
+      return Promise.resolve();
+    }
+    return service.setMute(value);
+  }
+
+  setHold(conversationId: string, value: boolean): Promise<any> {
+    const service = this.selectedImplementation;
+    if (!service || !service.isConnected) {
+      this.logger.info('Headset: No vendor headset connected [setHold]');
+      return Promise.resolve();
+    }
+    return service.setHold(conversationId, value);
+  }
+
+  endCall(conversationId: string, hasOtherActiveCalls?: boolean): Promise<any> {
+    const service = this.selectedImplementation;
+    if (!service || !service.isConnected) {
+      this.logger.info('Headset: No vendor headset connected [endCall]');
+      return Promise.resolve();
+    }
+    return service.endCall(conversationId, hasOtherActiveCalls);
+  }
+
+  endAllCalls(): Promise<any> {
+    const service = this.selectedImplementation;
+    if (!service || !service.isConnected) {
+      this.logger.info('Headset: No vendor headset connected [endAllCalls]');
+      return Promise.resolve();
+    }
+
+    return this.selectedImplementation.endAllCalls();
+  }
+
+  handleDeviceAnsweredCall(event: VendorEvent<EventInfo>): void | undefined {
+    console.log('event answered Call -> ', event);
+    if (event.vendor !== this.selectedImplementation) {
+      return;
+    }
+
+    this.logger.info('Headset: device answered the call');
+    this._headsetEvents$.next({ event: 'deviceAnsweredCall', payload: { ...event.body }});
+  }
+
+  handleDeviceRejectedCall(event: VendorConversationIdEvent): void | undefined {
+    if (event.vendor !== this.selectedImplementation) {
+      return;
+    }
+
+    this.logger.info('Headset: device rejected the call');
+    this._headsetEvents$.next({ event: 'deviceRejectedCall', payload: { conversationId: event.body.conversationId }});
+  }
+
+  handleDeviceEndedCall(event: VendorEvent<EventInfo>): void {
+    this.logger.info('Headset: device ended the call');
+    this._headsetEvents$.next({ event: 'deviceEndedCall', payload: { ...event.body } });
+    this._headsetEvents$.next({ event: 'loggableEvent', payload: { ...event.body } });
+  }
+
+  handleDeviceMuteStatusChanged(event: VendorEvent<MutedEventInfo>): void {
+    this.logger.info('Headset: device mute status changed: ', event?.body?.isMuted);
```

Comment:
do we need all the `?` here?

Reply:
Probably not, I think this was just me being extra (overly) careful.  In this event all these should exist